src/ovirtapi/transform.js

```diff
@@ -661,10 +655,79 @@ const StorageDomain = {
   toInternal ({ storageDomain }: { storageDomain: ApiStorageDomainType }): StorageDomainType {
     const permissions = Permissions.toInternal({ permissions: storageDomain?.permissions?.permission })
 
+    // frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#getDiskVolumeFormat
+    const diskAttributesForDiskType: {
+      ['thin' | 'pre']: { sparse: boolean, format?: 'raw' | 'cow' }
+    } = {
+      thin: { sparse: true },
+      pre: { sparse: false },
+    }
+
+    // frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#getVolumeType
+    const defaultDiskFormatToSparse: {
+      ['raw' | 'cow']: boolean
+    } = {
+      cow: true,
+      raw: true,
+    }
+
+    const storageType = storageDomain?.storage?.type ?? 'unknown'
+    let storageSubType
+    switch (storageType) {
+      case 'nfs':
+      case 'localfs':
+      case 'posixfs':
+      case 'glusterfs':
+      case 'glance':
+        storageSubType = 'file'
+        diskAttributesForDiskType.thin.format = 'raw'
+        diskAttributesForDiskType.pre.format = 'raw'
+        defaultDiskFormatToSparse.cow = true
+        defaultDiskFormatToSparse.raw = true
+        break
+
+      case 'fcp':
+      case 'iscsi':
+        storageSubType = 'block'
+        diskAttributesForDiskType.thin.format = 'cow'
+        diskAttributesForDiskType.pre.format = 'raw'
+        defaultDiskFormatToSparse.cow = true
+        defaultDiskFormatToSparse.raw = false
+        break
+
+      case 'cinder':
+      case 'managed_block_storage':
+        storageSubType = 'openstack'
+        diskAttributesForDiskType.thin.format = undefined
```

Comment:
unsported mode should be removed from the select box (when the domain is selected)

Reply:
`undefined` does not mean unsupported, it means unassigned

On the rest side, DiskFormat (cow | raw | null) maps to a VolumeFormat of (cow | raw | unassigned).